test/Microsoft.ML.Tests/Scenarios/Api/Estimators/TrainSaveModelAndPredict.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Learners;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public partial class ApiScenariosTests
+    {
+        /// <summary>
+        /// Train, save/load model, predict: 
+        /// Serve the scenario where training and prediction happen in different processes (or even different machines). 
+        /// The actual test will not run in different processes, but will simulate the idea that the 
+        /// "communication pipe" is just a serialized model of some form.
+        /// </summary>
+        [Fact]
+        public void New_TrainSaveModelAndPredict()
+        {
+            var dataPath = GetDataPath(SentimentDataPath);
+            var testDataPath = GetDataPath(SentimentTestPath);
+
+            using (var env = new TlcEnvironment(seed: 1, conc: 1))
+            {
+                // Pipeline.
+                var pipeline = new MyTextLoader(env, MakeSentimentTextLoaderArgs())
+                    .Append(new MyTextTransform(env, MakeSentimentTextTransformArgs()))
+                    .Append(new MySdca(env, new LinearClassificationTrainer.Arguments { NumThreads = 1 }, "Features", "Label"));
+
+                // Train.
+                var model = pipeline.Fit(new MultiFileSource(dataPath));
+
+                CompositeReader<IMultiStreamSource, ITransformer> loadedModel;
+                using (var file = env.CreateTempFile())
+                {
+                    // Save model. 
+                    using (var fs = file.CreateWriteStream())
+                        model.SavePipeline(env, fs);
+
+                    // Load model.
+                    loadedModel = CompositeReader.LoadPipeline(env, file.OpenReadStream());
```

Comment:
>loadedModel [](start = 20, length = 11)

should we use loadedModel for predictions?

Reply:
yup

---
In reply to: [211000998](https://github.com/dotnet/machinelearning/pull/688#discussion_r211000998) [](ancestors = 211000998)